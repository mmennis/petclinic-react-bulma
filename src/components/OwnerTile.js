import React from 'react'
import { Tile, Heading, Content, Card } from 'react-bulma-components'
import PropTypes from 'prop-types'

export default class OwnerTile extends React.Component {

    render() {
        return (
            <Tile size={3} renderAs="article" kind="child" notification color="light" paddingless={true} key={this.props.owner._id}>
                <Card paddingless={true} rounded="true" outlined="true" >
                    <Card.Header outlined="true" >
                        <Card.Header.Title>{this.props.owner.first_name} {this.props.owner.last_name}</Card.Header.Title>
                    </Card.Header>
                    <Card.Content paddingless={false} color="light">            
                        <Heading size={6} subtitle>{this.props.owner.pets.length} pets</Heading>
                        <Content >
                            <Heading marginless={true} paddingless={false} size={6}>Address:</Heading>
                            <p>{this.props.owner.address}<br/>{this.props.owner.city}, {this.props.owner.state}</p>
                        </Content>
                        <Content size="small">
                            <Heading marginless={true} paddingless={false} size={6}>Phone:</Heading >
                            <p>{this.props.owner.telephone}</p>
                        </Content>
                    </Card.Content>
                </Card>                
            </Tile>
        )
    }

}

OwnerTile.propTypes = {
    owner: PropTypes.object.isRequired
}