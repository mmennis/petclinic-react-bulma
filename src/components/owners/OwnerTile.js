import React from 'react'
import { Tile, Heading, Content, Card } from 'react-bulma-components'
import PropTypes from 'prop-types'
import OwnersDetail from './OwnersDetail'
import EditOwnerForm from './EditOwnerForm' 
import Pluralize from 'react-pluralize'

export default class OwnerTile extends React.Component {

    constructor(props) {
        super(props)

        this.state ={
            updated: false
        }
        this.handleOwnerUpdate = this.handleOwnerUpdate.bind(this)
    }

    handleOwnerUpdate = (owner) => {
        console.log(`Received an updated owner: ${JSON.stringify(owner)}`)
    }

    render() {
        return (
            <Tile size={3} renderAs="article" kind="child" notification color="light" paddingless={false} key={this.props.owner._id}>
                <Card paddingless={true} rounded="true" outlined="true" >
                    <Card.Header outlined="true" >
                        <Card.Header.Title>{this.props.owner.first_name} {this.props.owner.last_name}</Card.Header.Title>
                    </Card.Header>
                    <Card.Content paddingless={false} color="light">            
                        <Content size="small">
                            <p>{this.props.owner.city}, {this.props.owner.state}</p>
                        </Content>
                        <Heading size={4} subtitle><Pluralize singular={'pet'} count={this.props.owner.pets.length}/></Heading>
                    </Card.Content>
                    <Card.Footer>
                        <OwnersDetail owner={this.props.owner} modal={{closeOnBlur: true, showClose: true }}/>
                        <EditOwnerForm 
                            owner={this.props.owner} 
                            handleOwnerUpdate={this.handleOwnerUpdate} 
                            modal={{closeOnBlur: true, showClose: true }}
                        />
                    </Card.Footer>
                </Card>                
            </Tile>
        )
    }

}

OwnerTile.propTypes = {
    owner: PropTypes.object.isRequired
}