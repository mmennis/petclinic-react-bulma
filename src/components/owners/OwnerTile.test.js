import React from 'react'
import { create } from 'react-test-renderer'
import { render, fireEvent, getByTestId } from '@testing-library/react'
import OwnerTile from './OwnerTile'

const testOwner = {
    first_name: 'First',
    last_name: 'Last',
    address: '1234 Main St.',
    city: 'Cupertino',
    state: 'CA',
    telephone: '408-555-1212',
    pets: ['fido']
}

describe('OwnerTile', () => {
    it('should render as per the snapshot', () => {
        const callback = jest.fn()
        const ownerTile = create(<OwnerTile owner={testOwner} handleDeleteOwner={callback}/>)
        expect(ownerTile.toJSON()).toMatchSnapshot()
    })

    it('should have owner details in the output', () => {
        const callback = jest.fn()
        console.error = jest.fn()
        const { getByText } = render(<OwnerTile owner={testOwner}  handleDeleteOwner={callback}/>)
        const ownerFName = getByText(/First/i)
        expect(ownerFName).toBeInTheDocument()
        const ownerLName = getByText(/Last/i)
        expect(ownerLName).toBeInTheDocument()
        const city = getByText(/Cupertino/i)
        expect(city).toBeInTheDocument()
    })

    it('should log an error to the console if required owner props missing', () => {
        console.error = jest.fn();
        React.createElement(OwnerTile)
        expect(console.error).toHaveBeenCalledTimes(2);
    })

    it('should call the handleDeleteOwner callback when delete is clicked', () => {
        const callback = jest.fn()
        console.error = jest.fn()
        const { getByTestId } = render(<OwnerTile owner={testOwner}  handleDeleteOwner={callback}/>)
        fireEvent.click(getByTestId('delete-button'))
        expect(callback).toHaveBeenCalledTimes(1)
        expect(callback).toHaveBeenCalledWith(testOwner)
    })

})