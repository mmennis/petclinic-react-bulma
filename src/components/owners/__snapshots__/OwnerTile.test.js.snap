// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OwnerTile should render as per the snapshot 1`] = `
<article
  className="tile notification is-child is-3 is-light"
>
  <div
    className="card is-paddingless"
    outlined="true"
    rounded="true"
  >
    <div
      className="card-header"
      outlined="true"
    >
      <div
        className="card-header-title"
      >
        First
         
        Last
      </div>
    </div>
    <div
      className="card-content"
      color="light"
    >
      <div
        className="content is-small"
      >
        <p>
          Cupertino
          , 
          CA
        </p>
      </div>
      <h1
        className="subtitle is-4"
      >
        <span
          className={null}
          style={Object {}}
        >
          1 pet
        </span>
      </h1>
    </div>
    <div
      className="card-footer"
    >
      <div>
        <button
          className="is-small button"
          disabled={false}
          onClick={[Function]}
          tabIndex={0}
        >
          Details
        </button>
      </div>
      <div>
        <button
          className="is-small button"
          data-testid="modal-open"
          disabled={false}
          onClick={[Function]}
          tabIndex={0}
        >
          Update
        </button>
        <form
          data-testid="form"
          onSubmit={[Function]}
        />
      </div>
    </div>
  </div>
</article>
`;
